django/core/management/base.py

```diff
@@ -326,16 +328,16 @@ def run_from_argv(self, argv):
         handle_default_options(options)
         try:
             self.execute(*args, **cmd_options)
-        except Exception as e:
-            if options.traceback or not isinstance(e, CommandError):
+        except CommandError as e:
+            if options.traceback:
                 raise
 
             # SystemCheckError takes care of its own formatting.
             if isinstance(e, SystemCheckError):
                 self.stderr.write(str(e), lambda x: x)
             else:
                 self.stderr.write('%s: %s' % (e.__class__.__name__, e))
-            sys.exit(1)
```

Comment:
We shouldn't mix this cleanup with a new feature. Please move it to a separate commit also test is missing.